File: KalApp/app/src/main/java/com/example/kalapp/MenuActivity.java
package com.example.kalapp;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;

public class MenuActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_menu);
        LinearLayout logOut = findViewById(R.id.logoutbutton);
        // Enlaza con activity_menu.xml
        logOut.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Iniciar MainActivity
                Intent intent = new Intent(MenuActivity.this, MainActivity.class);
                startActivity(intent);
            }
        }
        );
    }
}

File: KalApp/app/src/main/java/com/example/kalapp/MenuActivity.kt
package com.example.kalapp

import android.content.Intent
import android.os.Bundle
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.ValueEventListener

enum class ProviderType {
    BASIC
}

class MenuActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu)

        val bundle = intent.extras
        val email = bundle?.getString("email")
        val provider = bundle?.getString("provider")
        setup(email ?: "", provider ?: "")
    }

    private fun setup(email: String, provider: String) {
        val logOut: LinearLayout = findViewById(R.id.logoutbutton)
        val nombreUsuario: TextView = findViewById(R.id.userName)
        val pesoUsuario: TextView = findViewById(R.id.userWeight)
        val alturaUsuario: TextView = findViewById(R.id.userHeight)

        title = "Inicio"

        // Obtén el userId del usuario actual
        val userId = FirebaseAuth.getInstance().currentUser?.uid

        if (userId != null) {
            val database = FirebaseDatabase.getInstance()
            val userRef = database.getReference("users").child(userId)

            // Lee los datos del usuario desde Firebase
            userRef.addListenerForSingleValueEvent(object : ValueEventListener {
                override fun onDataChange(snapshot: DataSnapshot) {
                    val userName = snapshot.child("name").value.toString()
                    nombreUsuario.text = userName
                    val userWeight = snapshot.child("weight").value.toString()
                    pesoUsuario.text = userWeight
                    val userHeight = snapshot.child("height").value.toString()
                    alturaUsuario.text = userHeight
                }

                override fun onCancelled(error: DatabaseError) {
                    // Maneja el error de lectura de datos
                    nombreUsuario.text = "Nombre no disponible"
                }
            })
        }

        logOut.setOnClickListener {
            FirebaseAuth.getInstance().signOut()
            val intent = Intent(this, MainActivity::class.java)
            startActivity(intent)
        }
    }
}

